From b9f259a0738509f7afc60832f456e57aa2c9c25b Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Mon, 22 Jun 2026 08:20:14 +0000
Subject: [PATCH] HID: core: zero-initialize the report buffer
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

jira VULN-40846
cve CVE-2024-50302
commit-author Jiri Kosina <jkosina@suse.com>
commit 177f25d1292c7e16e1199b39c85480f7f8815552

Since the report buffer is used by all kinds of drivers in various ways, let's
zero-initialize it during allocation to make sure that it can't be ever used
to leak kernel memory via specially-crafted report.

Fixes: 27ce405039bf ("HID: fix data access in implement()")
	Reported-by: Benoît Sevens <bsevens@google.com>
	Acked-by: Benjamin Tissoires <bentiss@kernel.org>
	Signed-off-by: Jiri Kosina <jkosina@suse.com>
(cherry picked from commit 177f25d1292c7e16e1199b39c85480f7f8815552)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
---
 drivers/hid/hid-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hid/hid-core.c b/drivers/hid/hid-core.c
index 9ff63f16845a..5de505c5f4c7 100644
--- a/drivers/hid/hid-core.c
+++ b/drivers/hid/hid-core.c
@@ -1648,7 +1648,7 @@ u8 *hid_alloc_report_buf(struct hid_report *report, gfp_t flags)
 
 	u32 len = hid_report_len(report) + 7;
 
-	return kmalloc(len, flags);
+	return kzalloc(len, flags);
 }
 EXPORT_SYMBOL_GPL(hid_alloc_report_buf);
 
-- 
2.52.0

